### PR TITLE
try to recheck and log new validator statuses

### PR DESCRIPTION
**Pull request by tzapu**
TODO(you): choose "part of" or "resolves" and the associated github issue #.

Part of #5002  

This adds the statuses of the public keys to metrics and also checks them every epoch and logs in stdout if the statuses have changed (assumes that the WaitForActivation stream will not block after the first time because it passed the initial wait.)

### CODE REVIEW

File: validator/client/validator.go [hunk 1]
```diff
@@ -256,7 +294,10 @@ func (v *validator) UpdateDuties(ctx context.Context, slot uint64) error {
 		log.Error(err)
 		return err
 	}
+	for k,v := range resp.Duties  {
+		log.Warnf("%+v\n%+v\n------", k, v)
```
Comment (rauljordan) [inline validator/client/validator.go]: What is the purpose of this warn log? Can we remove?

File: validator/client/validator.go [hunk 2]
```diff
@@ -49,6 +53,18 @@ type validator struct {
...
 
+var validatorStatusesGaugeVec = promauto.NewGaugeVec(
+	prometheus.GaugeOpts{
+		Namespace: "validator",
+		Name:      "statuses",
+		Help:      "validator statuses: 0 UNKNOWN, 1 DEPOSITED, 2 PENDING, 3 ACTIVE, 4 EXITING, 5 SLASHING, 6 EXITED",
+	},
+	[]string{
+		// validator pubkey
+		"pkey",
```
Comment (rauljordan) [inline validator/client/validator.go]: Can we rename to pubkey

File: validator/client/validator.go [hunk 3]
```diff
@@ -16,18 +16,21 @@ import (
...
 	ptypes "github.com/gogo/protobuf/types"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
-	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
 	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+
```
Comment (rauljordan) [inline validator/client/validator.go]:
```suggestion
```

File: validator/client/runner.go [hunk 4]
```diff
@@ -88,6 +89,12 @@ func run(ctx context.Context, v Validator) {
 			// Start fetching domain data for the next epoch.
 			if helpers.IsEpochEnd(slot) {
 				go v.UpdateDomainDataCaches(ctx, slot+1)
+
+				// update and log validator statues
```
Comment (rauljordan) [inline validator/client/runner.go]:
```suggestion
				// Update and log validator statuses.
```

**Comment by tzapu** [comment]
@rauljordan i m working on a cleaner version of doing this, maybe i should close this one and resubmit the other incarnation when it's ready (taking into account your feedback above, of course)
cheers

**Comment by tzapu** [comment]
closing this in favor of https://github.com/prysmaticlabs/prysm/pull/5069

**Pull request closed without merging**